pkg/controllers/nodeclaim/garbagecollection/controller.go

```diff
@@ -61,18 +61,18 @@ func (c *Controller) Reconcile(ctx context.Context) (reconcile.Result, error) {
 	if err != nil {
 		return reconcile.Result{}, fmt.Errorf("listing cloudprovider machines, %w", err)
 	}
-	managedRetrieved := lo.Filter(retrieved, func(nc *v1beta1.NodeClaim, _ int) bool {
-		return nc.Annotations[v1beta1.ManagedByAnnotationKey] != "" && nc.DeletionTimestamp.IsZero()
+	managedRetrieved := lo.Filter(retrieved, func(nc *corev1beta1.NodeClaim, _ int) bool {
+		return nc.Annotations[v1beta1.LabelNodeClass] != "" && nc.DeletionTimestamp.IsZero()
```

Comment:
Can we use the nodepool label? I think it should be the same, but I want to make sure we follow the convention in the upstream provider.
```suggestion
		return nc.Annotations[v1beta1.LabelNodePool] != "" && nc.DeletionTimestamp.IsZero()
```